WooCommerce/Classes/ViewRelated/Search/SearchViewController.swift

```diff
@@ -368,6 +386,13 @@ private extension SearchViewController {
                 self?.synchronizeSearchResults(with: query)
             }
     }
+
+    func configureSearchResync() {
+        searchUICommand.resynchronizeModels = { [weak self] in
+            guard let self = self else { return }
+            self.synchronizeSearchResults(with: self.searchQuery)
+        }
```

Comment:
❓ Will this override the variable set in the `searchUICommand`? If so the command will lose its initial variable, should we keep it as readonly in the protocol?

Reply:
this is the only place where the actual resync logic is implemented, I had to make it a callback variable in order for the command to call it when the results need to be resynced. I'm open to any other ideas!

also added a comment on the closure in the protocol in https://github.com/woocommerce/woocommerce-ios/pull/7739/commits/065b3b6ca9fe7f9338ba0fccc559436fa7ee034a